src/moe_pipeline.py

```
@@ -12,17 +12,25 @@ def __init__(self):
         num_experts = 4
-        expert_assignments = []
-        for _ in range(len(data)):
-            row = [random.random() for _ in range(num_experts)]
-            s = sum(row)
-            expert_assignments.append([x/s for x in row])
+        if override_assignments:
+            expert_assignments = override_assignments
+        else:
+            expert_assignments = []
+            for _ in range(len(data)):
+                row = [random.random() for _ in range(num_experts)]
```

Ruff (0.15.18)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)